pkg/tfvars/baremetal/baremetal.go

```diff
@@ -45,17 +45,19 @@ func init() {
 }
 
 // TFVars generates bare metal specific Terraform variables.
-func TFVars(libvirtURI, apiVIP, imageCacheIP, bootstrapOSImage, externalBridge, externalMAC, provisioningBridge, provisioningMAC string, platformHosts []*baremetal.Host, image, ironicUsername, ironicPassword, ignition string) ([]byte, error) {
+func TFVars(numControlPlaneReplicas int64, libvirtURI, apiVIP, imageCacheIP, bootstrapOSImage, externalBridge, externalMAC, provisioningBridge, provisioningMAC string, platformHosts []*baremetal.Host, image, ironicUsername, ironicPassword, ignition string) ([]byte, error) {
 	bootstrapOSImage, err := imageDownloader(bootstrapOSImage)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to use cached bootstrap libvirt image")
 	}
 
 	var masters, rootDevices, properties, driverInfos, instanceInfos []map[string]interface{}
 
-	for _, host := range platformHosts {
+	// Select the first N hosts as masters, excluding the workers
+	for i := 0; len(masters) < int(numControlPlaneReplicas); i++ {
```

Comment:
Can we add a check to avoid overrunning the slice, too, rather than relying on validation to ensure that the number of control-plane replicas is not too large?
```suggestion
	for i := 0; i < len(platformHosts) && len(masters) < int(numControlPlaneReplicas); i++ {
```

My personal preference would be the following. However, I am not opposed to what you have.
```suggestion
	for _, host := range platformHosts {
		if len(masters) >= int(numControlPlaneReplicas) {
			break
		}
```